common/web/lm-worker/src/main/model-compositor.ts

```diff
@@ -181,24 +130,35 @@ export default class ModelCompositor {
 
     // We want to dedupe before trimming the list so that we can present a full set
     // of viable distinct suggestions if available.
-    let deduplicatedSuggestionTuples = this.dedupeSuggestions(rawPredictions, context);
+    let deduplicatedSuggestionTuples = dedupeSuggestions(this.lexicalModel, rawPredictions, context);
 
     // Needs "casing" to be applied first.
     //
     // Will also add a 'keep' suggestion (with `.matchesModel = false`) matching
     // the current state of context if there is no such matching prediction.
-    this.processSimilarity(deduplicatedSuggestionTuples, context, transformDistribution[0]);
+    processSimilarity(this.lexicalModel, deduplicatedSuggestionTuples, context, transformDistribution[0]);
 
     // Section 3:  Sort the suggestions in display priority order to determine
     // which are most optimal, then auto-select based on the results.
     deduplicatedSuggestionTuples = deduplicatedSuggestionTuples.sort(tupleDisplayOrderSort);
-    this.predictionAutoSelect(deduplicatedSuggestionTuples);
+    predictionAutoSelect(deduplicatedSuggestionTuples);
 
     // Section 4: Trim down the suggestion list to the N (MAX_SUGGESTIONS) most optimal,
     // then cache and return the set of suggestions
 
     // Now that we've marked the suggestion to auto-select, we can finalize the suggestions.
-    const suggestions = this.finalizeSuggestions(deduplicatedSuggestionTuples, context, inputTransform);
+    const suggestions = finalizeSuggestions(
+      this.lexicalModel,
+      deduplicatedSuggestionTuples.splice(0, ModelCompositor.MAX_SUGGESTIONS),
```

Comment:
Is there a behavioural change here?

Reply:
There is not.  I've just reformatted these lines and ensured that the function still gets access to properties it had been using.  As the function is no longer a member of the `ModelCompositor` class, we have to explicitly pass them now.

Reply:
It's the `splice` that worried me :grin:

Reply:
Ah.  Still no - the original did a similar splice to chop off all the suggestions we wouldn't be returning to the user - should more than 12 even be found, of course.

Reply:
From `stable-17.0`:

https://github.com/keymanapp/keyman/blob/abfc5c268fa42a52189d7eaa01571cb15b261005/common/web/lm-worker/src/main/model-compositor.ts#L435